File: src/main/java/nu/peg/slack/pt/model/ConnectionRequest.java
package nu.peg.slack.pt.model;

import java.time.LocalTime;

import lombok.*;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ConnectionRequest {

    private Locations locations;
    private LocalTime time;
    private boolean isArrivalTime;

    public ConnectionRequest(Locations locations) {
        this.locations = locations;
        time = LocalTime.now();
        isArrivalTime = false;
    }

}

File: src/main/kotlin/nu/peg/slack/pt/model/ConnectionRequest.kt
package nu.peg.slack.pt.model

import java.time.LocalTime

import lombok.*

@Data
@NoArgsConstructor
@AllArgsConstructor
class ConnectionRequest(private val locations: Locations) {
    private val time: LocalTime
    private val isArrivalTime: Boolean

    init {
        time = LocalTime.now()
        isArrivalTime = false
    }

}
